plugins/file/plugin.go

```diff
@@ -113,10 +177,27 @@ func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
 
 // Handler4 handles DHCPv4 packets for the file plugin
 func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
-	// TODO check the MAC address in the request
-	//      if it is present in StaticRecords, forge a response
-	//      and stop processing.
-	return resp, true
+	record, ok := DHCPv4Records[req.ClientHWAddr.String()]
+	if !ok {
+		log.Printf("plugins/file: MAC address %s is new, leasing new IP address", req.ClientHWAddr.String())
+		rec, err := createIP(network)
+		if err != nil {
+			log.Error(err)
+			return nil, true
+		}
+		err = saveIPAddress(rec, req.ClientHWAddr)
+		if err != nil {
+			log.Printf("plugins/file: SaveIPAddress failed: %v", err)
+		}
+		DHCPv4Records[req.ClientHWAddr.String()] = rec
+		record = rec
+	}
+	resp.YourIPAddr = record.IP
+	resp.Options.Update(dhcpv4.OptIPAddressLeaseTime(LeaseTime))
+	resp.UpdateOption(dhcpv4.OptSubnetMask(network.Mask))
+	resp.UpdateOption(dhcpv4.OptRouter(network.IP))
```

Comment:
`network.IP` is something like `10.10.10.0` so this is probably not what you want to set as router.